h3/sf/doc/REFERENCE.md

```diff
@@ -0,0 +1,374 @@
+
+* `indexArray`: `ARRAY` of H3 cell indices as hexadecimal.
+* `resolution`: `INT` number between 0 and 15 with the [H3 resolution](https://h3geo.org/docs/core-library/restable).
+
+**Return type**
+
+`ARRAY`
+
+**Example**
+
+```sql
+SELECT SFCARTO.H3.UNCOMPACT(ARRAY_CONSTRUCT('847b59dffffffff'), 5);
+-- 857b59c3fffffff
+-- 857b59c7fffffff
+-- 857b59cbfffffff
+-- 857b59cffffffff
+-- 857b59d3fffffff
+-- 857b59d7fffffff
+-- 857b59dbfffffff
+```
+
+### TOPARENT
+
+{{% bannerNote type="code" %}}
+H3.TOPARENT(index, resolution)
+{{%/ bannerNote %}}
+
+**Description**
+
+Returns the H3 cell index of the parent of the given hexagon at the given resolution.
+
+* `index`: `STRING` H3 cell index as hexadecimal.
+* `resolution`: `INT` number between 0 and 15 with the [H3 resolution](https://h3geo.org/docs/core-library/restable).
+
+**Return type**
+
+`STRING`
+
+**Example**
+
+```sql
+SELECT SFCARTO.H3.TOPARENT('847b59dffffffff', 3);
+-- 837b59fffffffff
+```
+
+### TOCHILDREN
+
+{{% bannerNote type="code" %}}
+H3.TOCHILDREN(index, resolution)
+{{%/ bannerNote %}}
+
+**Description**
+
+Returns an array with the indexes of the children/descendents of the given hexagon at the given resolution.
+
+* `index`: `STRING` H3 cell index as hexadecimal.
+* `resolution`: `INT` number between 0 and 15 with the [H3 resolution](https://h3geo.org/docs/core-library/restable).
+
+**Return type**
+
+`ARRAY`
+
+**Example**
+
+```sql
+SELECT SFCARTO.H3.TOCHILDREN('837b59fffffffff', 4);
+-- 847b591ffffffff
+-- 847b593ffffffff
+-- 847b595ffffffff
+-- 847b597ffffffff
+-- 847b599ffffffff
+-- 847b59bffffffff
+-- 847b59dffffffff
+```
+
+### ISPENTAGON
+
+{{% bannerNote type="code" %}}
+H3.ISPENTAGON(index)
+{{%/ bannerNote %}}
+
+**Description**
+
+Returns `true` if given H3 index is a pentagon. Returns `false` otherwise, even on invalid input.
+
+* `index`: `STRING` H3 cell index as hexadecimal.
+
+**Return type**
+
+`BOOLEAN`
+
+**Example**
+
+```sql
+SELECT SFCARTO.H3.ISPENTAGON('837B59FFFFFFFFF');
+-- false
+```
+
+### DISTANCE
+
+{{% bannerNote type="code" %}}
+H3.DISTANCE(origin, destination)
+{{%/ bannerNote %}}
+
+**Description**
+
+Returns the **grid distance** between two hexagon indexes. This function may fail to find the distance between two indexes if they are very far apart or on opposite sides of a pentagon. Returns `null` on failure or invalid input.
+
+* `origin`: `STRING` H3 cell index as hexadecimal.
+* `destination`: `STRING` H3 cell index as hexadecimal.
+
+**Return type**
+
+`BIGINT`
+
+**Example**
+
+```sql
+SELECT SFCARTO.H3.DISTANCE('847b591ffffffff', '847b59bffffffff');
+-- 1
+```
+
+{{% bannerNote type="note" title="tip"%}}
+If you want the distance in meters use [ST_DISTANCE](https://cloud.google.com/bigquery/docs/reference/standard-sql/geography_functions#st_distance) between the cells ([ST_BOUNDARY](#st_boundary)) or their centroid.
```

Comment:
We should point to snowflake docs here: https://docs.snowflake.com/en/sql-reference/functions/st_distance.html

Reply:
Yes, thanks for the link.
We will also have to change "[Overview section](/spatial-extension-bq/spatial-indexes/overview/#h3)" at the beginning of the REFERENCE, in case links change when publishing snowflake doc.

Reply:
I guess we will have another section in the future `spatial-extension-sf` with everything Snowflake.